File: wallet/src/main/java/com/worldventures/wallet/service/command/profile/ChangedFields.java
package com.worldventures.wallet.service.command.profile;

import com.worldventures.wallet.domain.entity.SmartCardUserPhone;
import com.worldventures.wallet.domain.entity.SmartCardUserPhoto;

import org.immutables.value.Value;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
public interface ChangedFields {

   String firstName();

   String middleName();

   String lastName();

   @Nullable
   SmartCardUserPhoto photo();

   @Nullable
   SmartCardUserPhone phone();
}

File: wallet/src/main/java/com/worldventures/wallet/service/command/profile/ChangedFields.kt
package com.worldventures.wallet.service.command.profile

import com.worldventures.wallet.domain.entity.SmartCardUserPhone
import com.worldventures.wallet.domain.entity.SmartCardUserPhoto

data class ChangedFields(
      val firstName: String,
      val middleName: String,
      val lastName: String,
      val photo: SmartCardUserPhoto? = null,
      val phone: SmartCardUserPhone? = null
)